### DISCUSSION TITLE
Trouble referencing new row in 'WITH CHECK' RLS policy

### DISCUSSION THREAD

**Discussion by adamjr36** [supabase/supabase#20413] [Questions]
Hello,

I'm trying to create a policy on my trip_members table for insert. 
This table contains relevant columns 
1. trip_id, foreign key for trips
2. user_id, foreign key for users
3. role, specifying users role on that trip. 'owner' is one such value among others.

trip_members:
__id___trip_id___user_id____role___
|         |                |                  |                |
|         |                |                  |                |         
|         |                |                  |                |

I want my policy to allow users to insert if:
1. If a row already exists in trip_members where row.user_id = auth.uid() AND row.role == 'owner' AND row.trip_id = trip_id
OR
2. the new trip_id is not yet in trip_members

I wrote this with the help of many internet sources and the supabase AI query writer:
<img width="738" alt="Screenshot 2024-01-13 at 7 53 34 PM" src="https://github.com/supabase/supabase/assets/71796125/55a77ab7-d396-4f12-93ba-1f7c1ba9f1fd">

However, it autocorrects to this, which replaces trip_id, which I think should reference the NEW row being inserted, to tp.trip_id. This defeats the purpose of what I was trying to do and results in an error of infinite recursion:
<img width="732" alt="Screenshot 2024-01-13 at 7 55 41 PM" src="https://github.com/supabase/supabase/assets/71796125/b378bb58-10b5-4d22-a256-b7984c6e20fb">

Any help would be greatly appreciated!

**Comment by GaryAustin1** [comment]
I think the auto correct is OK as you are accessing the same table.

You will need to meed your select policy in both cases to use those selects though.  If you can't then you need to move the select part of your insert policy to a security definer function so that it bypasses RLS to do the select.
